rpc/grpc/server/services/blockservice/service.go

```diff
@@ -0,0 +1,130 @@
+package blockservice
+
+import (
+	context "context"
+	"fmt"
+
+	"github.com/cometbft/cometbft/libs/log"
+	cmtpubsub "github.com/cometbft/cometbft/libs/pubsub"
+	blocksvc "github.com/cometbft/cometbft/proto/tendermint/services/block/v1"
+	"github.com/cometbft/cometbft/store"
+	"github.com/cometbft/cometbft/types"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type blockServiceServer struct {
+	store    *store.BlockStore
+	eventBus *types.EventBus
+	logger   log.Logger
+}
+
+// New creates a new CometBFT version service server.
+func New(store *store.BlockStore, eventBus *types.EventBus, logger log.Logger) blocksvc.BlockServiceServer {
+	log := logger.With("module", "grpc-block-service")
+	return &blockServiceServer{
+		store,
+		eventBus,
+		log,
+	}
+}
+
+// GetByHeight implements v1.BlockServiceServer GetByHeight method
+func (s *blockServiceServer) GetByHeight(_ context.Context, req *blocksvc.GetByHeightRequest) (*blocksvc.GetByHeightResponse, error) {
+
+	logger := s.logger.With("endpoint", "GetByHeight")
+
+	var height int64
+
+	// retrieve the last height in the store
+	latestHeight := s.store.Height()
+
+	// validate height parameter, if height is 0 or
+	// the request is nil, then use the latest height
+	if req.Height == 0 {
+		height = latestHeight
+	} else if req.Height < 0 {
+		logger.Error("negative height request", "height", req.Height)
```

Comment:
Please capitalize the first letter of all log messages:

```suggestion
		logger.Error("Negative height request", "height", req.Height)
```

Same goes for the rest of this file.

It's only when we're constructing `error`s that we should not capitalize the first letter of the string, since errors can be nested. Log messages aren't nested though.